### ISSUE TITLE
spammy v235 "Detaching egress BPF program from cgroup failed: Invalid argument"

### ISSUE THREAD

**Issue by jwwalton** [systemd/systemd#7054]
When reloading and elsewhere I get
 
 ```
  % cat /proc/cgroups 
 #subsys_name	hierarchy	num_cgroups	enabled
 cpu	3	121	1
 cpuacct	3	121	1
 blkio	2	11	1
 devices	5	119	1
 pids	4	138	1
 
 ```
 ` systemd[1]: systemd 235 running in system mode. (+PAM -AUDIT -SELINUX -IMA -APPARMOR +SMACK -SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN default-hierarchy=hybrid)`

 
> Oct 11 00:48:22 foobar systemd[1]: Reloading.
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-logind.service failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-journald.service failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-udevd.service failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> Oct 11 00:48:22 foobar systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
> 
>

**Comment by mbiebl** [comment]
I can reproduce that problem on Debian sid.

**Comment by mbiebl** [comment]
seems to happen during boot and on ```daemon-reload``` or ```daemon-reexec```

**Comment by boucman** [comment]
do you know if your kernel has cgroup eBPF enabled ?

**Comment by mbiebl** [comment]
```
# grep BPF /boot/config-4.13.0-1-amd64 
# CONFIG_CGROUP_BPF is not set
CONFIG_BPF=y
CONFIG_BPF_SYSCALL=y
CONFIG_NETFILTER_XT_MATCH_BPF=m
CONFIG_NET_CLS_BPF=m
CONFIG_NET_ACT_BPF=m
CONFIG_BPF_JIT=y
CONFIG_LWTUNNEL_BPF=y
CONFIG_HAVE_EBPF_JIT=y
CONFIG_BPF_EVENTS=y
CONFIG_TEST_BPF=m
```

I guess that means no. I'd say systemd should handle this more gracefully and not log about this issue (or only once).

**Comment by poettering** [comment]
hmm, yeah, this shouldn't log at all

**Comment by poettering** [comment]
(but note that this is purely a cosmetic problem)

**Comment by djgera** [comment]
Cosmetic or not, until I did not enable CONFIG_CGROUP_BPF with systemd-235, systemd-networkd make things that depends on network-online.target or network.target on startup racy. In other words, services fail to start since network interface in not ready.

**Comment by Cherkah** [comment]
i get the same error in systemd's logs:

```
-Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-udevd.service failed: Invalid argument
-Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-logind.service failed: Invalid argument
-Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-localed.service failed: Invalid argument
-Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-hostnamed.service failed: Invalid argument
```
```
systemd 235
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN default-hierarchy=hybrid
```

```
# grep BPF /boot/config-4.13.0-1-amd64
# CONFIG_CGROUP_BPF is not set
CONFIG_BPF=y
CONFIG_BPF_SYSCALL=y
CONFIG_NETFILTER_XT_MATCH_BPF=m
CONFIG_NET_CLS_BPF=m
CONFIG_NET_ACT_BPF=m
CONFIG_BPF_JIT=y
CONFIG_LWTUNNEL_BPF=y
CONFIG_HAVE_EBPF_JIT=y
CONFIG_BPF_EVENTS=y
CONFIG_TEST_BPF=m 
```

what should i do? regards

**Comment by mbiebl** [comment]
@poettering should we downgrade that message to debug/drop it completely/log only once?

**Comment by poettering** [comment]
Fix waiting in #7298

**Comment by mbiebl** [comment]
Seems this issue is not really fixed. I've cherry-picked 93e93da5cc0499563710e3b77dcae3c342ed46b0 and ab8519c28b8904eb7e4f06d6b86614c7e6157f1a for Debian, where ```CONFIG_CGROUP_BPF=n```
After a reboot, I get:

```
journalctl -b | grep BPF
Nov 15 20:42:01 pluto systemd[1]: Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-udevd.service failed: Invalid argument
Nov 15 20:42:02 pluto systemd[1]: Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-logind.service failed: Invalid argument
Nov 15 20:42:02 pluto systemd[1]: Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-hostnamed.service failed: Invalid argument
Nov 15 20:42:06 pluto systemd[1]: Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/system.slice/systemd-localed.service failed: Invalid argument
```
After a ```systemctl daemon-reload``` I get
```
# journalctl -b | grep BPF | wc -l
160
```

**Comment by mbiebl** [comment]
The log is full with
```
Nov 15 20:42:43 pluto systemd[1]: Detaching egress BPF program from cgroup failed: Invalid argument
```

**Comment by poettering** [comment]
And what about current git? does that work properly?

If it doesn't any chance you can provide an strace output of bpf_firewall_supported() running on such a Debian kernel?

**Comment by mbiebl** [comment]
> And what about current git? does that work properly?

No, the log is still spammed with messages about egress BPF progams being attached/detached.

> If it doesn't any chance you can provide an strace output of bpf_firewall_supported() running on such a Debian kernel?

Can you give me some pointers how to create such a log?

**Comment by poettering** [comment]
> Can you give me some pointers how to create such a log?

```
strace -s 500 -f -o log build/test-bpf
```

from the source tree, after building should be good enough

**Comment by mbiebl** [comment]
@poettering  see attached strace log
[strace.txt](https://github.com/systemd/systemd/files/1503005/strace.txt)

**Comment by mbiebl** [comment]
(this is with git master v235-542-g2482f88d2) btw
Attached is a journalctl -r after a ```systemctl daemon-reload```
[log.txt](https://github.com/systemd/systemd/files/1503007/log.txt)

**Comment by poettering** [comment]
```
3494  bpf(BPF_MAP_CREATE, {map_type=BPF_MAP_TYPE_LPM_TRIE, key_size=8, value_size=8, max_entries=2, map_flags=BPF_F_NO_PREALLOC, inner_map_fd=0}, 48) = 14
3494  bpf(BPF_MAP_UPDATE_ELEM, {map_fd=14, key=0x7ffe99dc58c0, value=0x7ffe99dc58f0, flags=BPF_ANY}, 48) = 0
3494  bpf(BPF_MAP_UPDATE_ELEM, {map_fd=14, key=0x7ffe99dc58c0, value=0x7ffe99dc58f0, flags=BPF_ANY}, 48) = 0
3494  bpf(BPF_MAP_CREATE, {map_type=BPF_MAP_TYPE_LPM_TRIE, key_size=8, value_size=8, max_entries=2, map_flags=BPF_F_NO_PREALLOC, inner_map_fd=0}, 48) = 15
3494  bpf(BPF_MAP_UPDATE_ELEM, {map_fd=15, key=0x7ffe99dc58c0, value=0x7ffe99dc58f0, flags=BPF_ANY}, 48) = 0
3494  bpf(BPF_MAP_UPDATE_ELEM, {map_fd=15, key=0x7ffe99dc58c0, value=0x7ffe99dc58f0, flags=BPF_ANY}, 48) = 0
3494  bpf(BPF_PROG_LOAD, {prog_type=BPF_PROG_TYPE_CGROUP_SKB, insn_cnt=59, insns=0x55ae3727aa10, license="GPL", log_level=0, log_size=0, log_buf=0, kern_version=0, prog_flags=0}, 48) = 18
3494  open("/sys/fs/cgroup/unified/user.slice/user-1000.slice/user@1000.service/gnome-terminal-server.service/e2e158fd0569c22d/foo.service", O_RDONLY|O_DIRECTORY|O_CLOEXEC) = 19
3494  bpf(BPF_PROG_ATTACH, {target_fd=19, attach_bpf_fd=18, attach_type=BPF_CGROUP_INET_EGRESS, attach_flags=0}, 48) = -1 EINVAL (Invalid argument)
3494  close(19)                         = 0
3494  writev(2, [{iov_base="Attaching egress BPF program to cgroup /sys/fs/cgroup/unified/user.slice/user-1000.slice/user@1000.service/gnome-terminal-server.service/e2e158fd0569c22d/foo.service failed: Invalid argument", iov_len=190}, {iov_base="\n", iov_len=1}], 2) = 191
```

So, it appears your kernel supports cgroup/bpf just fine when it comes to the individual bits, but then refuses with EINVAL when we actually try to put it together: we can allocate the bpf program and its maps, it accepts our bpf code and everything, but then, when we actually what to attach the program to the cgroup it says EINVAL. And it's not clear at all why that might be... 

Not sure what I can recommend you to do. Is this a current kernel? How exactly are the relevant bpf and cgroup CONFIG_xyz options of the kernel set here?

**Comment by mbiebl** [comment]
@poettering See https://github.com/systemd/systemd/issues/7054#issuecomment-335793249

**Comment by mbiebl** [comment]
$ uname -a
Linux pluto 4.13.0-1-amd64 #1 SMP Debian 4.13.13-1 (2017-11-16) x86_64 GNU/Linux

**Comment by poettering** [comment]
ah, uh, so cgroup/bpf is turned off on your kernel after all?

So that means if CONFIG_CGROUP_BPF is off, but everything else we need is on, then `bpf(BPF_PROG_LOAD, {prog_type=BPF_PROG_TYPE_CGROUP_SKB, …})` will actually work, but then `bpf(BPF_PROG_ATTACH, {attach_type=BPF_CGROUP_INET_EGRESS, …})` won't? So we can create a cgroup bpf program, but we can't attach it to an actual cgroup?

What a broken API. Note entirely sure how we can detect cgroup/bpf support best then. I assumed that — if cgroup/bpf is turned off in the kernel – we couldn't create a cgroup/bpf program at all. But as it turns out we can...

Adding a check whether BPF_PROG_ATTACH works is much uglier, since we need to create a cgroup for that, and i'd rather not...

**Comment by wavexx** [comment]
On Mon, Nov 27 2017, Lennart Poettering wrote:
> So that means if CONFIG_CGROUP_BPF is off, but everything else we need

As an aside, but related, is there a reason why the Debian kernel didn't
enable CGROUP_BPF?

**Comment by poettering** [comment]
Suggested fix waiting in #7481. @mbiebl could you please give this a whirl to see if it works on debian kernels, too? thanks!

**Comment by mbiebl** [comment]
> As an aside, but related, is there a reason why the Debian kernel didn't
> enable CGROUP_BPF?

You'd probably need to ask the Debian kernel maintainers. There is
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=872560 tracking this issue.
It might make sense to follow up there and let them know that systemd is a potential user of CONFIG_CGROUP_BPF as well.

**Comment by cwrau** [comment]
Sorry for necrobumping this, I can create a new issue if you want, but I'm getting the same error on my arch;

```
Mar 05 13:32:23 steve systemd[549]: google-chrome.service: Attaching egress BPF program to cgroup /sys/fs/cgroup/user.slice/user-1000.slice/user@1000.service/app.slice/google-chrome.service failed: Invalid argument
```

Running

```
Linux steve 5.11.2-arch1-1 #1 SMP PREEMPT Fri, 26 Feb 2021 18:26:41 +0000 x86_64 GNU/Linux
```

And it doesn't seem to be a purely cosmetic problem, as all IP service stats are

```
IP: 0B in, 0B out
```

Although I can guarantee that some services are using the network, e.g. chrome from which I'm typing right now 😉
